visualization/app/codeCharta/ui/codeMap/codeMap.mouseEvent.service.ts

```diff
@@ -163,6 +168,13 @@ export class CodeMapMouseEventService
 	public onDocumentMouseDown(event) {
 		this.mouseOnLastClick = { x: event.clientX, y: event.clientY }
 		$(document.activeElement).blur()
+
+		if (event.button === ClickType.RightClick) {
+			this.$rootScope.$broadcast(CodeMapMouseEventService.MAP_RIGHT_CLICK_EVENT, {
```

Comment:
I took a look too, and fix wise I like it too!
But your branch seems out-of-date. On our current main branch, there is a subscriber for the onHideNodeContextMenu. You could fire that one right here by calling `NodeContextMenuController.broadcastHideEvent($rootScope)`
That way you would not need the right-click-subscriber